### PR TITLE
Fix dm_channel_id cache component check

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/impl/cache.py [hunk 1]
```diff
@@ -165,9 +165,11 @@ def set_dm_channel_id(
         channel: snowflakes.SnowflakeishOr[channels.PartialChannel],
         /,
     ) -> None:
-        user = snowflakes.Snowflake(user)
-        if self._is_cache_enabled_for(config.CacheComponents.EMOJIS) and user in self._user_entries:
-            self._dm_channel_entries[user] = snowflakes.Snowflake(channel)
+        user_id = snowflakes.Snowflake(user)
+        # XXX: Should this require the full user and cache that if not there?
```
Comment (FasterSpeeding) [inline hikari/impl/cache.py:169]: Rather than adding in-code TODOs can we migrate these to a github project where they can actually be tracked?

Comment (davfsa) [reply hikari/impl/cache.py:169]: Sure. This one in particular was to open a discussion and not be merged till removed (flake8 will complain about XXX but not TODO's)

Comment (davfsa) [reply hikari/impl/cache.py]: Removed the XXX and changed the code a bit :+1:

File: tests/hikari/impl/test_cache.py [hunk 2]
```diff
@@ -129,18 +129,6 @@ def test_set_dm_channel_id(self, cache_impl):
...
 
-    def test_set_dm_channel_id_when_user_not_known(self, cache_impl):
-        cache_impl.set_dm_channel_id(StubModel(43123123), StubModel(12222))
-
-        assert cache_impl._dm_channel_entries == {}
-
-    def test_set_dm_channel_id_when_not_enabled(self, cache_impl):
-        cache_impl._user_entries = collections.FreezableDict({43123123: object()})
-        cache_impl._settings.components = 0
-        cache_impl.set_dm_channel_id(StubModel(43123123), StubModel(12222))
```
Comment (FasterSpeeding) [inline tests/hikari/impl/test_cache.py:140]: presumably this test case still exists

File: hikari/impl/cache.py [hunk 3]
```diff
@@ -142,21 +142,33 @@ def clear(self) -> None:
...
         self, user: snowflakes.SnowflakeishOr[users.PartialUser], /
     ) -> typing.Optional[snowflakes.Snowflake]:
+        if not self._is_cache_enabled_for(config.CacheComponents.DM_CHANNEL_IDS):
+            return None
+
         return self._dm_channel_entries.get(snowflakes.Snowflake(user))
 
     def get_dm_channel_ids_view(self) -> cache.CacheView[snowflakes.Snowflake, snowflakes.Snowflake]:
+        if not self._is_cache_enabled_for(config.CacheComponents.DM_CHANNEL_IDS):
+            return cache_utility.EmptyCacheView()
```
Comment (FasterSpeeding) [inline hikari/impl/cache.py:170]: test coverage for these?

File: tests/hikari/impl/test_cache.py (hunk 2 above)
Comment (davfsa) [reply tests/hikari/impl/test_cache.py:140]: It has been moved down with all the other tests cases for disabled components

File: hikari/impl/cache.py (hunk 3 above)
Comment (davfsa) [reply hikari/impl/cache.py:170]: This has been moved down with all the other tests cases for disabled components

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**